CougSat1-ADCS/src/ADCSObjects.cpp

```diff
@@ -1,5 +1,53 @@
 #include "ADCSObjects.h"
 
-I2CSlave i2cBus(BUS_I2C0_SDA, BUS_I2C0_SCL);
+/**
+ * @brief Function for cdh communication thread
+ * 
+ */
+void ADCS::cdhThread()
+{
+    while (true)
+    {
+    if(cdh.messageReceived())
+    {
+        cdh.readCDH();
+    }
+    if(cdh.messageRequested())
+    {
+        cdh.writeCDH();
+    }
+    ThisThread::sleep_for(PERIOD_MS_POLLING_SLEEP);
+    }
+}
 
-CDH cdh(i2cBus);
\ No newline at end of file
+/**
+ * @brief Construct a new ADCS::ADCS object
+ * 
+ */
+#define TEST_IHU_ADDRESS 0xAC
+#define BUS_I2C0_SDA PinName(25)
+#define BUS_I2C0_SCL PinName(24)
+ADCS::ADCS() : cdh(TEST_IHU_ADDRESS, BUS_I2C0_SDA, BUS_I2C0_SCL)
+{
+    //monitor.set_priority(osPriorityNormal);
+    //cdhRead.set_priority(osPriorityNormal);
+    cdhRead.start(callback(this, &ADCS::cdhThread));
+    //monitor.start(callback(this, &ADCS::monitorThread));
+}
+
+/**
+ * @brief main function for ADCS object
+ * 
+ */
+void ADCS::initThread()
```

Comment:
Is this really the initThread??? It looks like adcs.run() to me and your init is in the constructor

Reply:
All functions that are being called as the main execution loop of a thread are labeled that way. There is always the main thread which will be kept in this function.

Reply:
I'm talking about the body of the function. You have a while true and comments that say inside and outside of main. Both of those don't mean initialize. Furthermore this does not start another main thread so that meaning doesn't work. So is this run(). Or does it actually fire off the other threads in which case start() would make more sense.